File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/BasicChoiceAlbumWrapper.java
package com.yanzhenjie.album.api;

import android.content.Context;

import androidx.annotation.IntRange;

/**
 * 混合选择器专用
 * 继承自：BasicChoiceWrapper
 * 作用：
 * 这个类是给【既能选图片、又能选视频、还能拍照/录像】的功能用的
 * 因为是混合模式，所以需要额外支持【相机录像参数】
 */
public abstract class BasicChoiceAlbumWrapper<Returner extends BasicChoiceAlbumWrapper, Result, Cancel, Checked> extends BasicChoiceWrapper<Returner, Result, Cancel, Checked> {
    // 相机录制视频的质量：0低质量 / 1高质量（默认1）
    protected int mQuality = 1;
    // 相机录制视频最大时长（秒）
    protected long mLimitDuration = Integer.MAX_VALUE;
    // 相机录制视频最大文件大小（字节）
    protected long mLimitBytes = Integer.MAX_VALUE;

    public BasicChoiceAlbumWrapper(Context context) {
        super(context);
    }

    /**
     * 设置相机录制视频的质量
     * @param quality 0 低质量 / 1 高质量
     */
    public Returner cameraVideoQuality(@IntRange(from = 0, to = 1) int quality) {
        this.mQuality = quality;
        return (Returner) this;
    }

    /**
     * 设置相机录制视频最大时长（秒）
     */
    public Returner cameraVideoLimitDuration(@IntRange(from = 1) long duration) {
        this.mLimitDuration = duration;
        return (Returner) this;
    }

    /**
     * 设置相机录制视频最大文件大小（字节）
     */
    public Returner cameraVideoLimitBytes(@IntRange(from = 1) long bytes) {
        this.mLimitBytes = bytes;
        return (Returner) this;
    }

}

File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/BasicChoiceAlbumWrapper.kt
package com.yanzhenjie.album.api

import android.content.Context
import androidx.annotation.IntRange

/**
 * Created by YanZhenjie on 2017/11/8.
 */
abstract class BasicChoiceAlbumWrapper<Returner : BasicChoiceAlbumWrapper<Returner, Result, Cancel, Checked>, Result, Cancel, Checked>(context: Context) : BasicChoiceWrapper<Returner, Result, Cancel, Checked>(context) {
    var mQuality = 1
    var mLimitDuration = Integer.MAX_VALUE.toLong()
    var mLimitBytes = Integer.MAX_VALUE.toLong()

    /**
     * Set the quality when taking video, should be 0 or 1. Currently value 0 means low quality, and value 1 means high quality.
     *
     * @param quality should be 0 or 1.
     */
    fun cameraVideoQuality(@IntRange(from = 0, to = 1) quality: Int): Returner {
        this.mQuality = quality
        return this as Returner
    }

    /**
     * Set the maximum number of seconds to take video.
     *
     * @param duration seconds.
     */
    fun cameraVideoLimitDuration(@IntRange(from = 1) duration: Long): Returner {
        this.mLimitDuration = duration
        return this as Returner
    }

    /**
     * Set the maximum file size when taking video.
     *
     * @param bytes the size of the byte.
     */
    fun cameraVideoLimitBytes(@IntRange(from = 1) bytes: Long): Returner {
        this.mLimitBytes = bytes
        return this as Returner
    }

}